src/api/api.controller.ts

```diff
@@ -1,12 +1,38 @@
-import { Controller, Get } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Get,
+  HttpCode,
+  Post,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import { CreateUserDto } from '../user/dtos/CreateUser.dto';
 import { ApiService } from './api.service';
+import { UserService } from '../user/user.service';
 
 @Controller('api')
 export class ApiController {
-  constructor(private readonly apiService: ApiService) {}
+  constructor(
+    private readonly apiService: ApiService,
+    private readonly userService: UserService,
+  ) {}
 
   @Get('hello-world')
   helloWorld(): string {
     return this.apiService.helloWorld();
   }
+
+  @Post('auth/create')
+  @HttpCode(201)
+  @UsePipes(new ValidationPipe())
```

Comment:
(praise): Wow. I really love this approach with request payload validation. Excellent!